packages/plugin-jsdocs/src/lib/config.ts

```diff
@@ -14,16 +18,19 @@ export const docCoveragePluginConfigSchema = z
       .describe(
         'List of audit slugs to evaluate. When specified, only these audits will be evaluated.',
       ),
-    sourceGlob: z
-      .array(z.string())
-      .default(['src/**/*.{ts,tsx}', '!**/*.spec.ts', '!**/*.test.ts'])
-      .describe('Glob pattern to match source files to evaluate.'),
+    patterns: patternsSchema,
   })
   .refine(data => !(data.skipAudits && data.onlyAudits), {
     message: "You can't define 'skipAudits' and 'onlyAudits' simultaneously",
     path: ['skipAudits', 'onlyAudits'],
   });
 
-export type DocCoveragePluginConfig = z.infer<
-  typeof docCoveragePluginConfigSchema
->;
+export const jsDocsPluginConfigSchema = z
+  .union([patternsSchema, jsDocsTargetObjectSchema])
+  .transform(target =>
+    typeof target === 'string' || Array.isArray(target)
+      ? { patterns: target }
+      : target,
+  );
+
+export type JsDocsPluginConfig = z.infer<typeof jsDocsPluginConfigSchema>;
```

Comment:
When you have a custom `.transform`, there's a subtle difference between `z.infer` and `z.input`. While `z.input` is the input type, while `z.infer` is the transformed output type.

Since `JsDocsPluginConfig` is used by the public `jsDocsPlugin` function, it should be the input type, so that callers can use the shorthand for patterns without getting type errors.

For using the transformed config in internal functions, you can create a separate type (e.g. called `JsDocsPluginTransformedConfig`, or even something shorter like `TransformedConfig` as it's not exported outside the package).

![image](https://github.com/user-attachments/assets/f57d81d1-c56c-4b04-9ffc-a0a37d3602ec)